carbon-design-system/carbon
# fix(link): exclude button elements from no-href styling

Addresses issue raised in this comment: https://github.com/carbon-design-system/carbon/issues/4705#issuecomment-555196276

There are situations, like with `BreadcrumbItem`, where `bx--link` class is added to an element.
If the child of `BreadcrumbItem` is a button, for example, then that button will get `bx--link` applied to it, and since the button won't have an `href`, it will get `:disabled` styling, including no pointer events. But that's problematic, as shown in https://github.com/carbon-design-system/carbon/issues/4705#issuecomment-555196276, because `:disabled` styling is being applied to an interactive button.

This PR proposes a small fix to exclude buttons from this `bx--link:not([href])` rule block. 

#### Changelog

**Changed**

- exclude button elements from `bx--link:not([href])` styling


#### Testing / Reviewing

A couple ways to test this...
* add a `Button` child to a `BreadcrumbItem` & see that it no longer has `disabled` styling and no pointer events
* add `bx--link` class to a `Button` component in the React storybook deployment for this PR (make sure you are adding the class to a `button` element)
